core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -4825,7 +4826,21 @@ Int32 NADefaults::validateFloat(const char *value, float &result,
 {
   Int32 n = -1;	// NT's scanf("%n") is not quite correct; hence this code-around
   sscanf(value, "%g%n", &result, &n);
-  if (n > 0 && value[n] == '\0') return TRUE;	// a valid float
+  if (n > 0 && value[n] == '\0') 
+    {
+      switch (attrEnum)
+        {
+        case HIVE_INSERT_ERROR_MODE:
```

Comment:
Not sure why this is needed. Why not follow the model, say, of validateUIntFrom0To5? (that is, add your own validateUIntFrom0To3)